### ISSUE TITLE
Upload To Media Root Under User's Own Username Folder

### ISSUE THREAD

**Issue by jaradc** [jazzband/sorl-thumbnail#641]
```
# models.py
from sorl.thumbnail import ImageField

def featured_images_folder(instance, filename):
    # docs to why this works: https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.FileField.upload_to
    return "{}/featured-images/{}".format(instance.user.username, filename)

featured_image = ImageField(upload_to=featured_images_folder, blank=True)
```
I am overriding the `ThumbnailBackend` to be more SEO-friendly (code found on MicroPyramid):

```
from sorl.thumbnail.conf import settings
from sorl.thumbnail.helpers import tokey, serialize
import os.path


class SEOThumbnailBackend(ThumbnailBackend):

    def _get_thumbnail_filename(self, source, geometry_string, options):
        """
        Computes the destination filename.
        """
        key = tokey(source.key, geometry_string, serialize(options))

        filename, _ext = os.path.splitext(os.path.basename(source.name))

        path = '%s/%s' % (key, filename)
        return '%s%s.%s' % (settings.THUMBNAIL_PREFIX, path, EXTENSIONS[options['format']])
```

The `settings.THUMBNAIL_PREFIX` is the reason files are being uploaded to a `cache/` directory, and not my desired `upload_to=` location.

If my goal is to have images uploaded to a specific user's username directory, how would I accomplish this? Uploads is the name of my media root:

```
MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'uploads')
```

Ex user with username "jason1" uploads an image. It gets uploaded to:
```
mediaroot / username / featured_images / put stuff here
uploads/jason1/featured_images/{ let sorl thumbnail do what it does here }
```

**Comment by jaradc** [comment]
TLDR;

How do I upload to a folder structure like this where the username is the name of my user:

uploads
 ----username1
--------featured_images
------------cache
----username2
--------featured_images
------------cache

I want to put the cache folder under a username within my mediaroot...

**Comment by skiv23** [comment]
You need to parse image source and get path to user directory and then use it as a prefix for your final result:


    class SEOThumbnailBackend(ThumbnailBackend):

        def _get_thumbnail_filename(self, source, geometry_string, options):
            """
            Computes the destination filename.
            """
            key = tokey(source.key, geometry_string, serialize(options))

            filename, _ext = os.path.splitext(os.path.basename(source.name))

            path = '%s/%s' % (key, filename)
            # trim filename to get original path to user's folder
            user_prefix_path = '/'.join(source.name.split('/')[:-1])
            return '%s/%s%s.%s' % (user_prefix_path, settings.THUMBNAIL_PREFIX, path, EXTENSIONS[options['format']])

![image](https://user-images.githubusercontent.com/5165402/90413931-cc554a00-e0b7-11ea-9b8b-61fe20206887.png)

**Comment by jaradc** [comment]
Thanks, I did something very similar.
